src/is/legacyError.ts

```diff
@@ -0,0 +1,13 @@
+import { ILegacyError } from '../types/error_types';
```

Comment:
There's a lot of mixing of file naming conventions going on here. I see some files `NamedLikeThis` others `namedLikeThis` and some others `named_liked_this`...at some point we should refactor for consistency. I like for files to be `named-liked-this` :sweat_smile:

Reply:
The rough naming convention I have been trying to follow is:
`NamedLikeThis` => exports a class of the same name
`namedLikeThis` => exports a function of the same name
`named_like_this` => exports several things

Definitely up to change, just not sure what the strategy should be.

Reply:
This looks like a very good standard to follow, despite the fact that it contradicts my preferences :sweat_smile: https://basarat.gitbook.io/typescript/styleguide#filename

Reply:
This too: https://github.com/airbnb/javascript#naming--filename-matches-export

Reply:
I will save this for a later PR, probably combined with a broader reorganizing of files to separate things out more.